plutus-contract-exe/src/Language/Plutus/Contract/Transaction.hs

```diff
@@ -0,0 +1,171 @@
+{-# LANGUAGE DeriveAnyClass         #-}
+{-# LANGUAGE DeriveGeneric          #-}
+{-# LANGUAGE DerivingStrategies     #-}
+{-# LANGUAGE FlexibleInstances      #-}
+{-# LANGUAGE FunctionalDependencies #-}
+{-# LANGUAGE MultiParamTypeClasses  #-}
+{-# LANGUAGE TemplateHaskell        #-}
+module Language.Plutus.Contract.Transaction(
+      UnbalancedTx
+    , emptyTx
+    , computeBalance
+    , inputs
+    , outputs
+    , forge
+    , requiredSignatures
+    , validityRange
+    , mergeWith
+    , toLedgerTx
+    -- * Constructing transactions
+    , unbalancedTx
+    , payToScript
+    , collectFromScript
+    , collectFromScriptFilter
+    ) where
+
+import           Control.Lens      (at, view, (^.))
+import qualified Control.Lens.TH   as Lens.TH
+import qualified Data.Aeson        as Aeson
+import qualified Data.Map          as Map
+import           Data.Maybe        (fromMaybe)
+import qualified Data.Set          as Set
+import           GHC.Generics      (Generic)
+
+import           Ledger            (Address, DataScript, PubKey, RedeemerScript, TxOut, TxOutRef, ValidatorScript)
+import qualified Ledger            as L
+import           Ledger.AddressMap (AddressMap)
+import qualified Ledger.Interval   as I
+import           Ledger.Slot       (SlotRange)
+import qualified Ledger.Tx         as Tx
+import           Ledger.Value      as V
+
+-- | An unsigned and potentially unbalanced transaction, as produced by
+--   a contract endpoint. See note [Unbalanced transactions]
+data UnbalancedTx = UnbalancedTx
+        { _Inputs             :: [(L.TxIn, V.Value)]
```

Comment:
Capitalizing these is weird.

Reply:
I take it that the reason this has values on it is so that you can calculate how much is needed to make it balance, but it's a bit odd. Do we actually need that ability? Can't we just hand it over to the wallet, which is presumably going to look stuff up in the index, etc?